File: labSocketsFestivalCSCompleted/labSocketsFestivalCSV2/ConsoleApplication1/persistence/CashierDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using model;

namespace persistence
{
    public class CashierDBRepository : ICashierRepository
    {
        private IDictionary<string, string> props;

        public CashierDBRepository(IDictionary<string, string> props)
        {
            this.props = props;
        }
        
        public void Save(Cashier entity)
        {
            throw new System.NotImplementedException();
        }

        public Cashier FindOne(int id)
        {
            throw new System.NotImplementedException();
        }

        public IEnumerable<Cashier> FindAll()
        {
            var connection = DbUtils.getConnection(props);
            IList <Cashier> cashiers= new List<Cashier>();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select * from cashiers";
                using (var dataR = command.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt16(0);
                        string username = dataR.GetString(1);
                        string password = dataR.GetString(2); 
                        Cashier cashier = new Cashier(id, username, password);
                        cashiers.Add(cashier);
                    }
                }
            }
            return cashiers;
        }

        public void Remove(int id)
        {
            throw new System.NotImplementedException();
        }

        public void Update(int id, Cashier e)
        {
            throw new System.NotImplementedException();
        }

        public Cashier GetCashierByUserPassword(string username, string password)
        {
            IDbConnection con = DbUtils.getConnection(props);
            Console.WriteLine("AIICIIII");
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from cashiers where username=@user and password=@pass";
                
                var paramUser = comm.CreateParameter();
                paramUser.ParameterName = "@user";
                paramUser.Value = username;
                comm.Parameters.Add(paramUser);

                var paramPass = comm.CreateParameter();
                paramPass.ParameterName = "@pass";
                paramPass.Value = password;
                comm.Parameters.Add(paramPass);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        Cashier cashier = new Cashier(dataR.GetInt32(0), dataR.GetString(1), dataR.GetString(2));
                        return cashier;
                    }
                }
            }

            return null;
        }
    }
}

File: labServiceUIJava/labServiceUIJava/src/main/java/com/example/labserviceuijava/repository/CashierDBRepository.java
package com.example.labserviceuijava.repository;

import com.example.labserviceuijava.model.Artist;
import com.example.labserviceuijava.model.Cashier;
import com.example.labserviceuijava.repository.exceptions.RepositoryException;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class CashierDBRepository implements CashierRepository{

    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public CashierDBRepository(Properties props) {
        logger.info("Initializing CashierRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public Cashier findOne(Integer integer) {
        return null;
    }

    @Override
    public Iterable<Cashier> findAll() {
        return null;
    }

    @Override
    public Cashier save(Cashier entity) {
        return null;
    }

    @Override
    public Cashier delete(Integer integer) {
        return null;
    }

    @Override
    public Cashier update(Cashier entity) {
        return null;
    }

    @Override
    public Cashier getCashierByUserPassword(String username, String password) {
        logger.traceEntry("find by username and password task {}");
        Connection con= dbUtils.getConnection();
        Cashier cashier=null;
        try(PreparedStatement preparedStatement=con.prepareStatement("select * from cashiers where username=? and password=?")){
            preparedStatement.setString(1,username);
            preparedStatement.setString(2,password);
            try(ResultSet resultSet=preparedStatement.executeQuery()){
                while(resultSet.next()){
                    cashier=new Cashier(resultSet.getString("username"),resultSet.getString("password"),resultSet.getInt("id"));
                }
            }
        }catch(SQLException e){
            logger.error(e);
            System.err.println("ERROR DB"+ e);
        }
        logger.traceExit(cashier);
        if(cashier==null)
            throw new RepositoryException("no cashier has been found with the given username and password");
        return cashier;
    }
}
